.github/workflows/github-actions-format-on-push.yml

```diff
@@ -0,0 +1,52 @@
+name: Format Code with pre commit trigger
+on:
+  # Triggers the workflow on push or pull request events
+  push:
+  pull_request:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  format:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: setup
+        run: |
+          sudo apt update 
+          sudo apt install -y clang clang-format
```

Comment:
Will this need to happen every time?  It might be better to make an image that has these pre-installed.

Reply:
we depend on GH images for this action. also I think GH has optimized this (or clang is already installed) since the step only takes ~10s.